### PR TITLE
Remove non application files from frontend code coverage

### PR DESCRIPTION
**Pull request by nickytonline**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

This PR cleans up our front-end coverage. There were files being covered that shouldn't be, e.g. barrel files and Storybook configurations.

## Related Tickets & Documents

#6126

## QA Instructions, Screenshots, Recordings

_Please replace this line with instructions on how to test your changes, as well
as any relevant images for UI changes._

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how accessibility is impacted and tested. For more info, check out the [Forem Accessibility Docs](https://docs.forem.com/frontend/accessibility)._

## Added tests?

- [ ] Yes
- [x] No, and this is why: The changes are test infrastructure changes for the front-end
- [ ] I need help with writing tests

## Added to documentation?

- [ ] [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/)
- [ ] README
- [x] No documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

Give a hoot! Don't pollute!

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 package.json
 app/assets/javascripts/lib/
+app/javascript/storybook-static/
```
Comment (nickytonline) [inline .eslintignore:3]: Came across this when running the tests to ensure this folder was omitted in test coverage. It is omitted from coverage, but it wasn't from eslint.

File: jest.config.js [hunk 2]
```diff
@@ -16,13 +16,19 @@ module.exports = {
...
+    '!app/javascript/**/*PropTypes.js',
+    '!./**/.eslintrc.js',
+    // Ignore Storybook configuration files
+    '!app/javascript/.storybook/**/*.{js,jsx}',
   ],
   coverageThreshold: {
     global: {
-      statements: 42,
-      branches: 38,
+      statements: 43,
```
Comment (nickytonline) [inline jest.config.js:28]: Bumped our coverage slightly as it went up when I cleaned up what was covered.

File: .eslintignore (hunk 1 above)
Comment (nickytonline) [inline .eslintignore:3]: Came across this when running the tests to ensure this folder was omitted in test coverage. It is omitted from coverage, but it wasn't from eslint.

File: jest.config.js [hunk 3]
```diff
@@ -16,13 +16,19 @@ module.exports = {
     '!**/__tests__/**',
     '!**/__stories__/**',
     '!app/javascript/storybook-static/**/*.js',
+    // We do not need code coverage on files that are prop types
+    // or eslint configuration files.
+    '!app/javascript/**/*PropTypes.js',
```
Comment (vaidehijoshi) [inline jest.config.js:21]: 👏

File: jest.config.js [hunk 4]
```diff
@@ -34,10 +40,10 @@ module.exports = {
   // picked up by jest if this folder is not excluded causing a false negative of a test suite failing.
   testPathIgnorePatterns: [
     '/node_modules/',
-    './config/webpack',
+    '<rootDir>/config/webpack',
```
Comment (vaidehijoshi) [inline jest.config.js:43]: Interesting, I didn't know about this...variable? (Is that what this is?)

I suppose we should have been using this from the get-go! :)

**Review by vaidehijoshi — APPROVED** [review]

**Review by benhalpern — APPROVED** [review]

**Pull request merged**